packages/elements/src/photon-multirx-form/index.tsx

```diff
@@ -40,414 +42,481 @@ export type Address = {
-    toastBuffer: 0
-  },
-  (
-    props: {
-      patientId?: string;
-      templateIds?: string;
-      templateOverrides?: TemplateOverrides;
-      prescriptionIds?: string;
-      hideSubmit: boolean;
-      hideTemplates: boolean;
-      enableOrder: boolean;
-      enableLocalPickup: boolean;
-      enableSendToPatient: boolean;
-      enableMedHistory: boolean;
-      mailOrderIds?: string;
-      pharmacyId?: string;
-      loading: boolean;
-      address?: Address;
-      weight?: number;
-      weightUnit?: string;
-      triggerSubmit: boolean;
-      setTriggerSubmit?: (val: boolean) => void;
-      toastBuffer: number;
-    },
-    options
-  ) => {
-    let ref: any;
-    const { store, actions } = createFormStore({
-      dispenseAsWritten: false,
-      patient: undefined,
-      treatment: undefined,
-      draftPrescriptions: [],
-      pharmacy: undefined,
-      errors: [],
-      address: undefined
+type PrescribeProps = {
+  patientId?: string;
+  templateIds?: string;
+  templateOverrides?: TemplateOverrides;
+  prescriptionIds?: string;
+  hideSubmit: boolean;
+  hideTemplates: boolean;
+  enableOrder: boolean;
+  enableLocalPickup: boolean;
+  enableSendToPatient: boolean;
+  enableMedHistory: boolean;
+  enableCombineAndDuplicate: boolean;
+  mailOrderIds?: string;
+  pharmacyId?: string;
+  loading: boolean;
+  address?: Address;
+  weight?: number;
+  weightUnit?: string;
+  triggerSubmit: boolean;
+  setTriggerSubmit?: (val: boolean) => void;
+  toastBuffer: number;
+  formStore?: any;
+  formActions?: any;
+};
+
+function PrescribeWorkflow(props: PrescribeProps) {
+  let ref: any;
+
+  const client = usePhoton();
+  const [showForm, setShowForm] = createSignal<boolean>(
+    !props.templateIds && !props.prescriptionIds
+  );
+  const [errors, setErrors] = createSignal<FormError[]>([]);
+  const [isLoading, setIsLoading] = createSignal<boolean>(true);
+  const [isEditing, setIsEditing] = createSignal<boolean>(false);
+  const [authenticated, setAuthenticated] = createSignal<boolean>(
+    client?.authentication.state.isAuthenticated || false
+  );
+  const [, recentOrdersActions] = useRecentOrders();
+
+  onMount(() => {
+    if (props.address) {
+      // if manually overriding address, update the store on mount
+      props.formActions.updateFormValue({
+        key: 'address',
+        value: props.address
+      });
+    }
+    if (ref) {
+      ref.addEventListener('photon-ticket-created-duplicate', () => {
+        clearForm(
+          props.formActions,
+          props.weight ? { notes: formatPatientWeight(props.weight, props?.weightUnit) } : undefined
+        );
+      });
+    }
+  });
+
+  createEffect(() => {
+    if (
+      !client?.authentication.state.isAuthenticated &&
+      !client?.authentication.state.isLoading &&
+      !client?.authentication.state.error
+    ) {
+      client?.authentication.login({ appState: { returnTo: window.location.pathname } });
+    }
+  });
```

Comment:
I know this already exists, but TBH I'm not sure what its supposed to be doing. I dont want to test it because it might break something but we should look into it

Reply:
TODO we should revisit this. if only we had a ticketing system...